### PR TITLE
Update arbiter bundle

### PR DESCRIPTION
**Pull request by connormanning**
- Add `az://` to the list of PDAL-recognized remote protocols
- Fix S3 globbing over HTTPS for bucket names which contain dots
- Add Azure SAS token support
- Some Azure env vars were aliased as both `AZ_ABC` and `AZURE_ABC`, made these aliases consistent throughout

### CODE REVIEW

**Review by hobu — APPROVED** [review]

File: vendor/arbiter/arbiter.cpp [hunk 1]
```diff
@@ -2655,6 +2665,17 @@ AZ::Config::Config(const std::string s, const std::string profile)
...
     , m_baseUrl(extractBaseUrl(s, m_service, m_endpoint, m_storageAccount))
 {
+    const std::string sasString = extractSasToken(s);
+    if (!sasString.empty())
+    {
+        const auto params = split(sasString, '&');
+        for (const auto& param : params)
+        {
+            const auto kv = split(param, '=');
+            m_sasToken[kv.at(0)] = kv.at(1);
```
Comment (abellgithub) [inline vendor/arbiter/arbiter.cpp:2675]: This needs some hardening. I'd rather see the result of split() checked for size and then use `[]`, but with `at`  you have to catch an exception.

File: vendor/arbiter/arbiter.cpp [hunk 2]
```diff
@@ -2693,6 +2714,10 @@ std::string AZ::Config::extractStorageAccount(
     {
         return *p;
     }
+    else if (auto p = env("AZ_STORAGE_ACCOUNT"))
```
Comment (abellgithub) [inline vendor/arbiter/arbiter.cpp:2717]: On line 2707, will `parse()` throw if the string isn't valid JSON?

File: vendor/arbiter/arbiter.cpp [hunk 3]
```diff
@@ -2720,6 +2749,25 @@ std::string AZ::Config::extractStorageAccessKey(
     return "";
 }
 
+std::string AZ::Config::extractSasToken(const std::string s)
+{
+    const json c(s.size() ? json::parse(s) : json());
```
Comment (abellgithub) [inline vendor/arbiter/arbiter.cpp:2754]: Same question here...

File: vendor/arbiter/arbiter.cpp [hunk 4]
```diff
@@ -2819,35 +2888,43 @@ bool AZ::get(
...
+
+    if (res->ok())
+    {
+        data = res->data();
         return true;
     }
     else
     {
-        std::cout << res.code() << ": " << res.str() << std::endl;
+        std::cout << res->code() << ": " << res->str() << std::endl;
```
Comment (abellgithub) [inline vendor/arbiter/arbiter.cpp:2927]: This seems like it should throw, log, or something other than just random output.

File: vendor/arbiter/arbiter.cpp [hunk 5]
```diff
@@ -2738,6 +2786,14 @@ std::string AZ::Config::extractService(
     {
         return *p;
     }
+    else if (auto p = env("AZ_SERVICE"))
```
Comment (abellgithub) [inline vendor/arbiter/arbiter.cpp:2789]: Someone please tell me why `auto` is better than `const char *`.

File: vendor/arbiter/arbiter.cpp [hunk 6]
```diff
@@ -2720,6 +2749,25 @@ std::string AZ::Config::extractStorageAccessKey(
     return "";
 }
 
+std::string AZ::Config::extractSasToken(const std::string s)
```
Comment (abellgithub) [inline vendor/arbiter/arbiter.cpp:2752]: Is there a reason this isn't a reference, given that it's already const? Some threading thing?

**Comment by hobu** [comment]
oops, I thought this was ready to merge. Please push to the branch again with your additional updates @connormanning and reopen a PR.

**Pull request merged**